Userland/Libraries/LibGfx/QOIWriter.cpp

```diff
@@ -0,0 +1,212 @@
+/*
+ * Copyright (c) 2022, the SerenityOS developers.
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include "QOIWriter.h"
+#include "QOICommon.h"
+#include <AK/ByteBuffer.h>
+#include <AK/Math.h>
+#include <AK/MemoryStream.h>
+#include <AK/TypedTransfer.h>
+#include <LibGfx/Bitmap.h>
+
+namespace Gfx {
+
+template<typename Stream = ByteBuffer>
+class QOIWriterState : public QOIState {
+public:
+    ErrorOr<void> start(u32 width, u32 height, bool has_alpha_channel)
+    {
+        if (m_pixels_to_encode != INVALID_PIXELS_TO_ENCODE)
+            return Error::from_string_literal("Trying to start an already initialized QOI writer"sv);
+
+        m_pixels_to_encode = static_cast<u64>(width) * static_cast<u64>(height);
+        VERIFY(m_pixels_to_encode != INVALID_PIXELS_TO_ENCODE);
+
+#if 0
```

Comment:
Is this dead? Delete?

Reply:
It is a reminder, that I need to submit a patch to add `ByteBuffer::reserve`

Reply:
You probably want `ByteBuffer::try_ensure_capacity()`.